src/backend/distributed/commands/index.c

```diff
@@ -590,13 +605,30 @@ PreprocessReindexStmt(Node *node, const char *reindexCommand,
 		Oid relationId = ReindexStmtFindRelationOid(reindexStatement, false);
 		MemoryContext relationContext = NULL;
 		Relation relation = NULL;
+		LOCKMODE lockmode = IsReindexWithParam_compat(reindexStatement, "concurrently") ?
```

Comment:
now that we have multiple use of this lock decision, maybe we should create a new function - -similar to `GetCreateIndexRelationLockMode`, like `GetReindexIndexRelationLockMode` or such